### PR TITLE
wip: Debian packaging

### PR DESCRIPTION
**Pull request by samuelkarp**
TODO:
- [x] Figure out how to switch from `3.0 (native)` to `3.0 (quilt)`, since that's preferred for projects that are not Debian-only
- [x] Figure out how to properly remove `ecr-login/vendor` for the Debian build
- [x] Fix `statically-linked-binary` lintian error
  - The "upstream" build script builds a fully-static binary and disables cgo
  - Alternately, switch back to `dh_auto_build` and additionally specify ldflags for `version.Version` and `version.GitCommitSHA`
- [ ] Fix `no-copyright-file` lintian error
- [x] Fix `subdir-in-usr-bin` lintian error -> binary is installed in the wrong place
- [x] Fix `syntax-error-in-debian-changelog` lintian warning
- [x] Fix `no-section-field` lintian warning
- [x] Write a manpage and fix `binary-without-manpage` lintian warning

/cc @nmeyerhans

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Source: amazon-ecr-credential-helper
+# Section
+Priority: extra
```
Comment (nmeyerhans) [inline debian/control]: Use 'optional', 'extra' is deprecated in current policy versions.
https://www.debian.org/doc/debian-policy/ch-archive.html#priorities

File: debian/control [hunk 2]
```diff
@@ -0,0 +1,27 @@
+Source: amazon-ecr-credential-helper
+# Section
```
Comment (nmeyerhans) [inline debian/control]: Pick something from https://www.debian.org/doc/debian-policy/ch-archive.html#sections

File: debian/files [hunk 3]
```diff
@@ -0,0 +1,2 @@
+amazon-ecr-credential-helper_0.1.0_amd64.buildinfo - extra
```
Comment (nmeyerhans) [inline debian/files]: `debian/files` shouldn't be committed at all. https://www.debian.org/doc/debian-policy/ch-source.html#generated-files-list-debian-files

File: debian/.debhelper/generated/amazon-ecr-credential-helper/installed-by-dh_install [hunk 4]
```diff
@@ -0,0 +1 @@
+./bin/local/docker-credential-ecr-login
```
Comment (nmeyerhans) [inline debian/.debhelper/generated/amazon-ecr-credential-helper/installed-by-dh_install]: The `.debhelper` directory shouldn't be committed.

File: debian/rules [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+GITCOMMIT_SHA := $(shell git rev-parse --short HEAD)
+package_root := "github.com/awslabs/amazon-ecr-credential-helper/ecr-login"
+
+%:
+	dh $@ --buildsystem=golang --with=golang
+
+override_dh_auto_build:
+	dh_auto_build -- -ldflags "-X $(package_root)/version.Version=$(VERSION) -X $(package_root)/version.GitCommitSHA=$(GITCOMMIT_SHA)"
+	mkdir -p bin/local
+	cp obj-$(DEB_HOST_GNU_TYPE)/bin/docker-credential-ecr-login bin/local
```
Comment (nmeyerhans) [inline debian/rules:19]: Do you actually need to be doing any of the stuff on lines 18-19?

File: debian/files (hunk 3 above)
Comment (samuelkarp) [reply debian/files]: Thanks, I wasn't sure about whether `debian/files` and `debian/.debhelper` should be committed.

File: debian/rules (hunk 5 above)
Comment (samuelkarp) [reply debian/rules:19]: Can I use `obj-$(DEB_HOST_GNU_TYPE)` in `amazon-ecr-credential-helper.install`?  If so, I don't need these lines.

Comment (nmeyerhans) [reply debian/rules:19]: You can't. My (incomplete) suggestion was more to override the `dh_auto_install` rule and install the build artifacts into the package in a custom rule. It seems somewhat strange to copy the files to a new location just so you can move them out of it.

Comment (samuelkarp) [reply debian/rules:19]: Should I do that instead of using `amazon-ecr-credential-helper.install`?

**Comment by samuelkarp** [comment]
I think all the major parts are done.  What remains is basically project structure:

* Where the `debian` folder and its files should be stored (my preference is under `packaging`, like I've done for the Amazon Linux PR)
* How to convert to `3.0 (quilt)` (I think I can make a tarball in the Makefile, but I'm not wild about taking a dependency on `$(cat VERSION)` for the filename)
* How to remove the `vendor` folder properly (should I just exclude it from the tarball?)

After that, I need to learn how to submit this to Debian.

**Comment by samuelkarp** [comment]
Closing and reopening against the upstream repo.

**Pull request closed without merging**